week2/homework/src/index.js

```diff
@@ -1,3 +1,118 @@
 'use strict';
+let fs = require('fs');
+let order = process.argv[2];
+let orderDefinition = process.argv[3];
+let orderDefinition2 = process.argv[4];
 
-// TODO: Write the homework code in this file
+fs.readFile('./toDo.json', 'utf8', function (error, data) {
+  if (error) {
+    console.log(error);
+  }
+  else {
+    let myToDoList = JSON.parse(data);
+
+    function help() {
+      fs.readFile('./help.txt', 'utf8', function (error, data) {
+        if (error) {
+          console.log(error);
+        }
+        else {
+          console.log(data);
+        }
+      }
+      );
+    };
+
+    function list() {
+      function isEmpty(myToDoList) {
+        if (Object.values(myToDoList).length === 0) {
+          return true;
+        }
+        return false;
+      }
+      if (isEmpty(myToDoList)) {
+        console.log('You have not any plan in your To Do List');
+      }
+      else {
+        console.log(myToDoList);
+      }
+    };
+
+    function add(orderDefinition, orderDefinition2) {
+      let toDoDate = orderDefinition2.toLocaleDateString();
+      myToDoList[Object.keys(myToDoList).length + 1] = orderDefinition + ' ' + toDoDate;
+      fs.writeFile(
+        './toDo.json',
+        JSON.stringify(myToDoList),
+        (error) => {
+          if (error) { console.log(error); }
+        });
+    };
+
+    function remove(orderDefinition) {
+      if (orderDefinition > 0) {
+        delete myToDoList[orderDefinition];
+        fs.writeFile(
+          './toDo.json',
+          JSON.stringify(myToDoList),
+          (error) => {
+            if (error) { console.log(error); }
+          });
+      }
+      else {
+        console.log('The parameter you entered as index is invalid');
+      }
+    };
+
+    function reset() {
+      fs.writeFile(
+        './toDo.json',
+        '{}',
+        (error) => {
+          if (error) { console.log(error); }
+        });
+    };
+
+    function update(orderDefinition, orderDefinition2) {
+      if (orderDefinition > 0) {
+        myToDoList[orderDefinition] = orderDefinition2;
+        fs.writeFile(
+          './toDo.json',
+          JSON.stringify(myToDoList),
+          (error) => {
+            if (error) { console.log(error); }
+          });
+      }
+      else {
+        console.log('The parameter you entered as index is invalid');
+      }
+    };
+
+    switch (order) {
+      case 'help':   // help shows help section
+        help();
+        break;
+      case undefined:   // no command shows help section
+        help();
+        break;
```

Comment:
since the 2 `case`s above do the exact same thing you can rewrite this code as:
```js
case 'help':
case undefined:
  help();
  break;
```

Reply:
Also, keep in mind that the `undefined` case is handled by the `default` case you have at the end of the `switch` :)

Reply:
Understood, corrected